### ISSUE TITLE
Sourcing sharness.sh returns 1 instead of 0

### ISSUE THREAD

**Issue by chriscool** [mlafeldt/sharness#47]
Since https://github.com/mlafeldt/sharness/commit/b65226ffea8f1d8365bb1cfda760462642d39432 sourcing sharness.sh returns 1 instead of 0 previously.

This is because the above commit added the following line at the end of sharness.sh:

```
test -n "$TEST_LONG" && test_set_prereq EXPENSIVE
```

If "$TEST_LONG" is empty, the exit code of sourcing sharness.sh is then the exit code of `test -n "$TEST_LONG"` which is 1.

**Comment by chriscool** [comment]
I think a simple fix would be replacing the above line with:

```
test -z "$TEST_LONG" || test_set_prereq EXPENSIVE
```

Will push a PR with something like the above.

CC @moy

**Comment by moy** [comment]
Why not come back to a plain "if/then"? I find it more readable than your fix, which is somehow a double-negation (I read it as "unless `$TEST_LONG` is not set, run test_set_prereq ...").

**Comment by mlafeldt** [comment]
Another option would be to end the script with `:`, which is commonly done in init scripts exactly for that purpose.

**Comment by chriscool** [comment]
Yeah, ending the script with `:` is a good idea and can be done instead or on top of PR #48.
I think I will do that soon.

About using `if/then/fi` I don't think it is enough, because people might be later tempted to simplify it again into using just `&&`.

**Comment by chriscool** [comment]
Fixed by PR #49 (sharness: properly ensure it ends with code 0).
